projects/plugins/jetpack/json-endpoints/class.wpcom-json-api-site-settings-endpoint.php

```diff
@@ -270,6 +270,70 @@ protected function get_cast_option_value_or_null( $option_name, $cast_callable )
 		return call_user_func( $cast_callable, $option_value );
 	}
 
+	/**
+	 * Used for the WooCommerce address options. This function returns the option value if it exists, else it looks
+	 * for a default address on the domain contact information. If that does not exist it returns an empty string.
+	 *
+	 * @param string $key The option key to retrieve a value.
+	 *
+	 * @return string The requested address part value or an empty string.
+	 */
+	protected function get_woocommerce_address( $key ) {
+		$address_part = get_option( $key );
+
+		// If there's any value set in the database, return it. Even if it's null or an empty string.
+		// False means no value for this option was found in the database.
+		if ( false !== $address_part ) {
+			return $address_part;
+		}
+
+		// If no value is set in the database, return the corresponding address part from the user's domain contact
+		// address. If that does not exist return an empty string.
+		$address_part = '';
+		if ( defined( 'IS_WPCOM' ) && IS_WPCOM ) {
+			$blog_owner_user_id = wpcom_get_blog_owner( get_current_blog_id() );
+			$address            = Domain_Contact_Information_Mapper::find_by_user_id( $blog_owner_user_id );
+
+			if ( $address ) {
+				switch ( $key ) {
+					case 'woocommerce_store_address':
+						$address_part = $address->get_address_1();
+						break;
+					case 'woocommerce_store_address_2':
+						$address_part = $address->get_address_2();
+						break;
+					case 'woocommerce_store_city':
+						$address_part = $address->get_city();
+						break;
+					case 'woocommerce_default_country':
+						// WooCommerce expects this setting to be $country_code:$region_code like US:CA.
+						$country_code = $address->get_country_code();
+						$state_name   = $address->get_state_name(); // Note this is 'state_name' not region_code.
+
+						// Get the list of regions and region_codes supported by WooCommerce.
+						$regions_by_country_code = require_once __DIR__ . '/woocommerce/states.php';
```

Comment:
I added states.php file here so this method will work. But, I'm not sure this is the best location for this and the countries.php files. Where should these 2 files live?

Reply:
I removed this feature code from this PR and drafted an issue for this in the backlog. We can convert that to an issue and follow up there.